include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -4491,6 +4492,47 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
 
 
 
+template <int dim,
+          int n_components_,
+          typename Number,
+          bool is_face,
+          typename VectorizedArrayType>
+inline Point<dim, VectorizedArrayType>
+FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
+  quadrature_point(const unsigned int q) const
+{
+  AssertIndexRange(q, this->n_quadrature_points);
+  Assert(this->quadrature_points != nullptr,
+         internal::ExcMatrixFreeAccessToUninitializedMappingField(
+           "update_quadrature_points"));
+
+  // Cartesian/affine mesh: only first vertex of cell is stored, we must
+  // compute it through the Jacobian (which is stored in non-inverted and
+  // non-transposed form as index '1' in the jacobian field)
+  if (is_face == false &&
+      this->cell_type <= internal::MatrixFreeFunctions::affine)
+    {
+      Assert(this->jacobian != nullptr, ExcNotInitialized());
+      Point<dim, VectorizedArrayType> point = this->quadrature_points[0];
+
+      const Tensor<2, dim, VectorizedArrayType> &jac = this->jacobian[1];
+      if (this->cell_type == internal::MatrixFreeFunctions::cartesian)
+        for (unsigned int d = 0; d < dim; ++d)
+          point[d] += jac[d][d] * static_cast<Number>(
+                                    this->descriptor->quadrature.point(q)[d]);
+      else
+        for (unsigned int d = 0; d < dim; ++d)
+          for (unsigned int e = 0; e < dim; ++e)
+            point[d] += jac[d][e] * static_cast<Number>(
+                                      this->descriptor->quadrature.point(q)[e]);
+      return point;
+    }
+  else
+    return this->quadrature_points[q];
```

Comment:
Looking at this code, I wonder if we can now move this to `FEEvaluationData`, where we have all the other access functions?

Reply:
Done!